### PR TITLE
Increase cache assets file format version to 11 in 7.0.1xx

### PR DESCRIPTION
**Pull request by ocallesp**
## Description

The cache assets file was modified in https://github.com/dotnet/sdk/pull/27580 to store two new values `DebugSymbolsFiles` and `ReferenceDocumentationFiles` and hence the format version should be increased.

Original Issue: https://github.com/dotnet/sdk/issues/1458

## Customer Impact

The cache assets file format changed and this can cause builds to fail with weird errors difficult to debug.

## Regression?

- [x] Yes
- [ ] No

Builds can fail

## Risk

- [] High
- [X] Medium
- [] Low

Medium risk. Even though this can affect all builds, doing a restore fixes this.

## Verification

- [x] Manual (required)
- [x] Automated

### CODE REVIEW

**Review by dsplaisted — APPROVED** [review]

**Comment by dsplaisted** [comment]
@ocallesp Don't actually merge this yet, we will need to get approval for it.

**Comment by ocallesp** [comment]
> @ocallesp Don't actually merge this yet, we will need to get approval for it.

Who should approve this change ?

**Pull request merged**
